_rules/links-with-identical-names-and-context-serve-equivalent-purpose-fd3a94.md

```diff
@@ -42,7 +42,8 @@ This rule applies to any set of two or more [HTML or SVG elements][] for which a
 - the elements are in the same [web page (HTML)][]; and
 - the elements are [included in the accessibility tree][included in the accessibility tree]; and
 - the elements have [matching][] [accessible names][accessible name] that are not empty (`""`); and
-- have the same [programmatically determined link context][].
+- the elements have the same [programmatically determined link context][]; and
+- the purpose of the elements is not [ambiguous to users in general](https://www.w3.org/TR/WCAG21/#dfn-ambiguous-to-users-in-general). 
```

Comment:
I'm having second thoughts about this. The definition is 

> the purpose cannot be determined from the link and all information of the Web page presented to the user simultaneously with the link

I don't think this is objective. I agree it is unambiguous, so we might rewrite the rule in a way that moves this condition from the applicability to the expectation (and update the examples accordingly). 

What do you think?

Reply:
@carlosapaduarte ,

I agree, even more considering that the definition of "ambiguous to users in general" is far away from our definition of "programmatically determined link context".

On it

Reply:
@carlosapaduarte,

I've tried with multiple solutions, but for all of them, removing "not ambiguous to users in general" from the applicability, makes the expectation even more complicated to understand.

In addition, the rule title is "Links with identical accessible names and same context serve equivalent purpose", but removing the "not ambiguous to users in general" from the applicability makes the title no longer valid; in fact links with identical accessible names and same context (but ambiguous to users in general) might serve different purposes, resulting in a pass for the rule that contrasts with the title.

What do you think?

Reply:
My suggestion would be to update the current expectation to something like 

> For the links in each set of target elements, one of the following is true:
> - The links are not [ambiguous to users in general](https://www.w3.org/TR/WCAG21/#dfn-ambiguous-to-users-in-general) and, when followed, resolve to the [same resource][] or to [equivalent resources](#equivalent-resource); or
> - The links are [ambiguous to users in general](https://www.w3.org/TR/WCAG21/#dfn-ambiguous-to-users-in-general)

Reply:
The we don't even need the exception in the first bullet:
> either
> * the links resolve to the same resource; or
> * the links are ambiguous

Reply:
- Links not ambiguous to users in general has been removed from Applicability
- Added expectation as per suggestions
- Slightly changed the background note about ambiguous links to reflect new applicability
- Moved Inapplicable example n.8 to passed example 9 (changing a bit the description).